mk/install-build-tools.sh

```diff
@@ -128,6 +128,12 @@ case $target in
     libc6-dev-riscv64-cross \
     qemu-user
   ;;
+--target=riscv32gc-unknown-linux-gnu)
+  use_clang=1
+  install_packages \
+    #https://github.com/riscv-collab/riscv-gnu-toolchain#installation-newliblinux-multilib
```

Comment:
I see that this target is quite difficult to support. I read "If you need riscv32-linux, then yes you will need to build it." from https://github.com/riscv-collab/riscv-gnu-toolchain/issues/808.

It seems like my niave and underinformed guess that riscv32gc-unknown-linux-gnu would be the easiest riscv32 target to support was wrong.

Is riscv32gc-unknown-linux-gnu the actual target you are trying to support, or were you just trying to follow my suggestion? If you don't need riscv32gc-unknown-linux-gnu but instead need a different target, I think we should focus on the one that's actually useful.

Reply:
Thank you for taking the time for this analysis.

Ok, don't forget you ask for it not me.... don't kill the messenger. 

Now the bad news: 
our target [riscv32imac-esp-espidf](https://doc.rust-lang.org/nightly/rustc/platform-support/esp-idf.html) is a [Tier 3](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-3). Meaning target is supported but not built, distributed, or guaranteed to work. 

So, for our target to work we need to have the [ESP-IDF Toolchain](https://docs.espressif.com/projects/esp-idf/en/v5.1.2/esp32/get-started/linux-macos-setup.html#standard-toolchain-setup-for-linux-and-macos) and then have a config.toml thing similar to this :

```rust
[build]
target = "riscv32imac-esp-espidf"

[target.riscv32imac-esp-espidf]
linker = "/tools/riscv32-esp-elf/esp-12.2.0_20230208/riscv32-esp-elf/bin/ld"

[unstable]
build-std = ["std", "panic_abort"]

[env]
MCU="esp32c6"
ESP_IDF_VERSION = "v5.1.1"
TARGET_CC = "/tools/riscv32-esp-elf/esp-12.2.0_20230208/riscv32-esp-elf/bin/riscv32-esp-elf-cc"
```

For the ring crate build.

As example let's run the test case ed25519_tests:

```bash
cargo build --test ed25519_tests
```

Result...

![image](https://github.com/briansmith/ring/assets/142290531/b138fa4a-233d-4302-9029-637240c5790b)


So, I am open to suggestions on your side....when you have the time of course.  :frowning_face: